turbopack/crates/turbo-tasks-backend/src/backend/storage.rs

```diff
@@ -1096,6 +1096,38 @@ macro_rules! update_count {
     };
 }
 
+macro_rules! update_count_and_get {
+    ($task:ident, $key:ident $input:tt, -$update:expr) => {{
+        let update = $update;
+        let mut new = 0;
+        $crate::backend::storage::update!($task, $key $input, |old: Option<_>| {
+            let old = old.unwrap_or(0);
+            new = old - update;
+            (new != 0).then_some(new)
+        });
+        new
+    }};
+    ($task:ident, $key:ident $input:tt, $update:expr) => {
+        match $update {
+            update => {
+                let mut new = 0;
+                $crate::backend::storage::update!($task, $key $input, |old: Option<_>| {
+                    let old = old.unwrap_or(0);
+                    new = old + update;
+                    (new != 0).then_some(new)
+                });
+                new
+            }
+        }
+    };
```

Comment:
Strange that one of them uses `match` and the other doesn't

Reply:
Fixing in a follow up pr